### PR TITLE
refactor(Registry): Refactor out all Configuration related APIs

### LINKED ISSUE #37: Refactor to remove configuration APIs
Per the Registry Refactoring design (https://github.com/edgexfoundry/edgex-docs/blob/master/design/adr/0001-Registy-Refactor.md) the configuration  APIs will be removed and put in new go-mod-configuration module

### PR DESCRIPTION
**Pull request by lenny-intel**
Remove all abstract Configuration related code/unit tests/mocks
Implement Unregister addressing #20 
Addressed #14 
Other minor code tweaks

Closes #37 , #20 & #14

### CODE REVIEW

**Comment by lenny-intel** [comment]
recheck

**Comment by lranjbar** [comment]
The old verify pipeline has been removed now to stop the double builds. Need to rebase off the master branch and push.

**Comment by lenny-intel** [comment]
> LGTM Lenny. Where does all the refactored out config elated API go? To bootstrap, contracts or new go-mod-config package?

@jpwhitemn , it is now here: https://github.com/edgexfoundry-holding/go-mod-configuration

File: internal/pkg/consul/mock_consul.go [hunk 1]
```diff
@@ -18,15 +18,12 @@ package consul
 
 import (
 	"encoding/json"
+	consulapi "github.com/hashicorp/consul/api"
```
Comment (tsconn23) [inline internal/pkg/consul/mock_consul.go]: Should be at bottom of import list

**Comment by lenny-intel** [comment]
> LGTM.
> 
> I would suggest not merging this until go-mod-configuration is brought in from holding because the current `master` of edgex-go will not build against this PR. Recommended sequence is:
> 
> ```
> - Bring go-mod-configuration out of holding
> - Publish initial version for above
> - Refactor edgex-go against versioned go-mod-configuration and local version of this PR via replace directive in go.mod
> - Merge this PR
> - Modify edgex-go go.mod to point to new version of go-mod-registry, remove replace directive
> - Create edgex-go PR
> ```

Agree mostly.. ;-) Once go-mod-configuration is moved out of holding these will be integrated into go-mod-bootstrap.

**Comment by lenny-intel** [comment]
@tsconn23 , draft PRs are now available for using these changes. This PR needs to be merged before those PRs can be moved from Draft.
https://github.com/edgexfoundry/go-mod-bootstrap/pull/34
https://github.com/edgexfoundry/edgex-go/pull/2317

**Comment by tsconn23** [comment]
recheck

**Pull request merged**